authorization/authorization.go

```diff
@@ -0,0 +1,161 @@
+package authorization
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"log"
+	"net/http"
+	"time"
+
+	jwtmiddleware "github.com/auth0/go-jwt-middleware"
+	jwt "github.com/dgrijalva/jwt-go"
+)
+
+type Handshake struct {
+	nonce      string
+	providerID string
+}
+
+// Outstanding handshakes.
+var handshakes map[string]Handshake
```

Comment:
I don't think we should save these globally.

Reply:
Suggestions on where to save them? We can't really use a server struct if we use the middleware libraries we're using now, so I wasn't sure where to put them.

Reply:
We can use a server struct. You have to use a closure to capture the struct with the method you're passing to the middleware. Example:

Current:
```
router.Handle("/providers/{id}", authMiddleware.Handler(getProviderHandler)).Methods("GET")
```

New:
```
server := &Metaserver{}
router.Handle("/providers/{id}", authMiddleware.Handler(server.getProviderHandler)).Methods("GET")
```

Reply:
Ah, didn't think of that. I'll update it to do that.